src/components/icon/icon.component.tsx

```diff
@@ -19,7 +21,10 @@ export type LegacyIconTypes =
   | "success"
   | "messages";
 
-export interface IconProps extends Omit<StyledIconProps, "type">, MarginProps {
+export interface IconProps
+  extends Omit<StyledIconProps, "type">,
+    MarginProps,
+    TagProps {
```

Comment:
suggestion (non-blocking): `data-component` is private/ ignored in the TagProps interface, it would not complain if a value is passed but wouldn't do anything so maybe it's worth just omitting it to avoid this

```suggestion
    Omit<TagProps, "data-component"> {
```

Reply:
Happy to add this in 👍🏼 It may be necessary though to set "icon" as the default value for `data-role` within `Icon` itself, due to scenarios where it is deeply nested in the React DOM tree and we cannot directly set the attribute's value:

```tsx
// in file-upload-status.component.tsx...

const fileLink = linkProps ? (
    <Link {...linkProps}>{filename}</Link> // 👈 Cannot set data-role of Icon directly
  ) : (
    /* falsy case */
  );
```

Reply:
Might be worth discussing if we need both the `data-element` and `data-role` tags going forward, as they seem to be used interchangeably by the tests

Reply:
I don't mind this approach, the alternative is to set it in the Link component like below but I'm not overly bothered which approach we take

```
 <Icon
          type={icon}
          disabled={disabled}
          ariaLabel={removeAriaLabelOnIcon ? undefined : ariaLabel}
          tooltipMessage={tooltipMessage}
          tooltipPosition={tooltipPosition}
          data-role={`${icon}-link-icon`}
        />